File: 7. Reverse Integer/Reverse_Integer.py
import sys
class Solution:
    def reverse(self, x: int) -> int:
        n = x
        r = 0
        c = 1
        if n < 0:
            n *= -1
            c = -1
        while(n>0):
            a  = n % 10
            r = r*10 + a
            n //= 10
        if (r < -2**31 or r > 2**31 - 1):
            return 0
        #print(2**31)
        return r*c

File: 7. Reverse Integer/Reverse_Integer.cpp
class Solution {
public:
    int reverse(int n) 
    {
        int max = INT_MAX, min = INT_MIN;
        long long r = 0;
        long long int x = n;
        int s = 1;
        if(x < 0)
        {
            s = -1;
            x = -x;
        }
        while(x > 0)
        {
            int d = x % 10;
            r = r * 10 + d;
            x /= 10;
        }
        r *= s;
        if(r < min or r > max)
        {
            return 0;
        }
        return r;
    }
};
